minidns-dane-java7/src/main/java/de/measite/minidns/dane/DaneExtendedTrustManager.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dane;
+
+import de.measite.minidns.AbstractDNSClient;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedTrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.logging.Logger;
+
+public class DaneExtendedTrustManager extends X509ExtendedTrustManager {
+    private final static Logger LOGGER = Logger.getLogger(DaneExtendedTrustManager.class.getName());
+
+    private final X509TrustManager base;
+    private final DaneVerifier verifier;
+
+    public static void inject() {
+        inject(new DaneExtendedTrustManager());
+    }
+
+    public static void inject(DaneExtendedTrustManager trustManager) {
```

Comment:
I'm not sure if this is the right thing to do (at least here). Users of this library should rather ingect it on their own. I also think that wiping all other trust managers would be bad.

How about moving this to a readme or examples?

PS: I think this class is awesome. Are you sure this does not work on Android? I'd expect XMPP clients to love this code.